parts/kubernetesmastercustomscript.sh

```diff
@@ -92,6 +93,43 @@ function ensureKubectl() {
     fi
 }
 
+function setNetworkPlugin () {
+    sed -i "s/^KUBELET_NETWORK_PLUGIN=.*/KUBELET_NETWORK_PLUGIN=${1}/" /etc/default/kubelet
+}
+
+function setDockerOpts () {
+    sed -i "s#^DOCKER_OPTS=.*#DOCKER_OPTS=${1}#" /etc/default/kubelet
+}
+
+function configNetworkPolicy() {
+    if [[ ! -z "${APISERVER_PRIVATE_KEY}" ]]; then
+        # on masters
+        ADDONS="calico-configmap.yaml calico-daemonset.yaml"
+        ADDONS_PATH=/etc/kubernetes/addons
+        CALICO_URL="https://raw.githubusercontent.com/projectcalico/calico/a4ebfbad55ab1b7f10fdf3b39585471f8012e898/v2.0/getting-started/kubernetes/installation/hosted/k8s-backend-addon-manager"
+        if [[ "${NETWORK_POLICY}" = "calico" ]]; then
+            # download calico yamls
+            for addon in ${ADDONS}; do
+                curl -sSL --retry 12 --retry-delay 10 "${CALICO_URL}/${addon}" > "${ADDONS_PATH}/${addon}"
+            done
+        else
+            # download calico yamls
+            for addon in ${ADDONS}; do
+                rm -f "${ADDONS_PATH}/${addon}"
+            done
+        fi
+    else
+        # on agents
+        if [[ "${NETWORK_POLICY}" = "calico" ]]; then
```

Comment:
@lxpollitt thank you, I assumed as much but I don't understand how these things compose. My understanding is that the existing CNI plugin being used, `kubenet`, is essential for making networking + cloudprovider work. But I guess somehow removing `kubenet` and replacing with `calico` just works still somehow? Maybe calico knows how to do the bridge reconfiguration that `kubenet` does?

There's a further extension to this question, but I don't really know if I can talk about it here yet. Are you or @simonswine available for a call anytime today? Can even just be voice? I just want to understand a tiny bit better before signing off.